src/routes/api/hotels.route.js

```diff
@@ -0,0 +1,155 @@
+ *             type: object
+ *             properties:
+ *               status:
+ *                 type: string
+ *               message:
+ *                 type: string
+ *               data:
+ *                 type: object
+ *                 properties:
+ *                   id:
+ *                     type: integer
+ *                   locationId:
+ *                     type: string
+ *                   name:
+ *                     type: string
+ *                   image:
+ *                     type: string
+ *                   street:
+ *                     type: string
+ *                   description:
+ *                     type: string
+ *                   services:
+ *                     type: string
+ *                   updatedAt:
+ *                     type: string
+ *                   createAt:
+ *                     type: string
+ *     responses:
+ *       201:
+ *         description: success
+ */
+router.post(
+  '/hotels',
+  verifyUser,
+  authorizeTravelAdmin,
+  fileService.upload('image'),
+  validation,
+  catchErrors(hotels.createHotel)
+);
+
+/**
+ * @swagger
+ *
+ * /hotels/{hotelId}/rooms:
+ *   post:
+ *     summary: Add Hotel rooms
+ *     description: This allows travel admins to add hotel rooms
+ *     tags:
+ *       - Accommodation
+ *     requestBody:
+ *       content:
+ *         multipart/form-data:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               name:
+ *                 type: integer
+ *               image:
+ *                 type: string
+ *                 format: binary
+ *               description:
+ *                 type: string
+ *               type:
+ *                 type: string
+ *               cost:
+ *                 type: string
+ *     produces:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               status:
+ *                 type: string
+ *               message:
+ *                 type: string
+ *               data:
+ *                 type: object
+ *                 properties:
+ *                   id:
+ *                     type: integer
+ *                   hotelId:
+ *                     type: string
+ *                   name:
+ *                     type: string
+ *                   image:
+ *                     type: string
+ *                   type:
+ *                     type: string
+ *                   description:
+ *                     type: string
+ *                   cost:
+ *                     type: integer
+ *                   status:
+ *                     type: string
+ *                   updatedAt:
+ *                     type: string
+ *                   createAt:
+ *                     type: string
+ *     responses:
+ *       201:
+ *         description: success
+ */
+router.post(
+  '/hotels/:hotelId/rooms',
+  verifyUser,
+  fileService.upload('image'),
+  authorizeTravelAdmin,
+  validation,
```

Comment:
If authorisation or validation fails how do we handle already uploaded image?

Reply:
yeah, that may cause problems, I'm going to do validation before upload

Reply:
I have been trying to do validation after upload. However, validations were failing. because validation middleware requires access to `req.body`. 
For this case, we are sending `multipart/formdata`  and I found that body-parser `does not handle multipart bodies, due to their complex and typically large nature.` 

I found that we need to pass upload middleware which is able to parse `multipart/formdata` and pass `req.body` to the next middleware.

[](https://www.npmjs.com/package/body-parser#readme)
![image](https://user-images.githubusercontent.com/15703806/69990833-22e11f00-154f-11ea-8a14-30ebc06b6d1d.png)

Reply:
We can go with `multer` since we are already using it for file upload.